File: solutions/1512. Number of Good Pairs/1512_1.cpp
// Time complexity: O(n^2)
// Space complexity: O(1)
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int count = 0;
        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] == nums[j]) count += 1;
            }
        }
        return count;
    }
};

File: solutions/1512. Number of Good Pairs/1512_1.cs
// Time complexity: O(n^2)
// Space complexity: O(1)
public class Solution {
    public int NumIdenticalPairs(int[] nums) {
        int ans = 0;
        for (int i = 0; i < nums.Length; ++i) {
            for (int j = i + 1; j < nums.Length; ++j) {
                if (nums[i] == nums[j]) {
                    ++ans;
                }
            }
        }
        return ans;
    }
}